### PR TITLE
[Doc]add indexes overview

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
Why I'm doing:

What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [X] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/en/table_design/indexes/Bitmap_index.md [hunk 1]
```diff
@@ -2,19 +2,26 @@
...
 
-A bitmap index can help improve the query performance on a given column. If a query hits a sort key column, StarRocks efficiently returns the query result by using the [prefix index](../table_design/Sort_key.md). However, the prefix index entry for a data block cannot exceed 36 bytes in length. If you want to improve the query performance on a column, which is not used as a sort key, you can create a bitmap index for the column.
+A bitmap index can help improve the query performance on a given column. If a query hits a sort key column, StarRocks efficiently returns the query result by using the [prefix index](./indexes_overview.md#prefix-indexes). However, the prefix index entry for a data block cannot exceed 36 bytes in length. If you want to improve the query performance on a column, which is not used as a sort key, you can create a bitmap index for the column.
+
+Bitmap indexes are generally suitable for columns with high cardinality. Bitmap indexes are a good choice when  Bitmap indexes can exhibit high selectivity, and its filtering effect (number of data rows filtered by the Bitmap index/total number of data rows) is lower than one in ten thousand.
```
Comment (amber-create) [inline docs/en/table_design/indexes/Bitmap_index.md:13]: image
extra space

File: docs/zh/sql-reference/sql-statements/data-manipulation/test.yaml [hunk 2]
```diff
@@ -0,0 +1,5939 @@
+---
```
Comment (amber-create) [inline docs/zh/sql-reference/sql-statements/data-manipulation/test.yaml:1]: Why no such a cn doc is available? pls check if this is the way it should be.

File: docs/zh/table_design/StarRocks_table_design.md [hunk 3]
```diff
@@ -119,11 +119,11 @@ StarRocks 提供两种分桶方式：
 
 除了基本的数据类型，如数值、日期和字符串类型，StarRocks 还支持复杂的半结构化数据类型，包括 ARRAY、JSON、MAP、STRUCT。
 
-### 索引
+### [索引](./indexes/indexes_overview.md)
 
 索引是一种特殊的数据结构，相当于数据的目录。查询条件命中索引列时，StarRocks 能够快速定位到满足条件的数据的位置。
 
-StarRocks 内置一级索引，是根据组成排序键的列的前缀组成的前缀索引。也支持用户手动创建二级索引，以提高查询效率，包括 Bitmap 和 Bloom Filter 索引。
+StarRocks 提供内置索引，包括前缀索引，Ordinal 索引和ZoneMap 索引。也支持用户手动创建索引，以提高查询效率，包括 Bitmap 和 Bloom Filter 索引。
```
Comment (amber-create) [inline docs/zh/table_design/StarRocks_table_design.md:126]: image
1. 用顿号。
2. 前面加空格。

File: docs/en/table_design/StarRocks_table_design.md [hunk 4]
```diff
@@ -118,11 +118,11 @@ StarRocks provides two bucketing methods:
 
 In addition to basic data types such as NUMERIC, DATE, and STRING, StarRocks supports complex semi-structured data types, including ARRAY, JSON, MAP, and STRUCT.
 
-### Index
+### [Index](./indexes/indexes_overview.md)
 
 An index is a special data structure and is used as a pointer to data in a table. When the conditional columns in queries are indexed columns, StarRocks can swiftly locate the data that meets the conditions.
 
-StarRocks provides a built-in primary index, which is a prefix index composed of the prefixes of the table's sort key columns. StarRocks also allows you to create secondary indexes, including Bitmap and Bloom Filter indexes, to further enhance query efficiency.
+StarRocks provides built-in indexes, including Prefix indexes. StarRocks also allows users to create indexes, including Bitmap indexes and Bloom Filter indexes, to further enhance query efficiency.
```
Comment (amber-create) [inline docs/en/table_design/StarRocks_table_design.md:125]: image
I guess you forgot "Ordinal indexes and ZoneMap indexes".

Comment (amber-create) [inline docs/en/table_design/StarRocks_table_design.md:125]: There's a question: do we support only Prefix indexes, Ordinal indexes, and ZoneMap indexes as built-in indexes?
If so, "including" is not a good choice of word. Use "StarRocks provides built-in indexes: Prefix indexes, Ordinal indexes, and ZoneMap indexes."
The same for "StarRocks also allows users to create indexes, including Bitmap indexes and Bloom Filter indexes ..."

File: docs/zh/table_design/indexes/Bitmap_index.md [hunk 5]
```diff
@@ -8,11 +8,18 @@ displayed_sidebar: "Chinese"
 
 Bitmap 索引是一种使用 bitmap 的特殊数据库索引。bitmap 即为一个 bit 数组，一个 bit 的取值有两种：0 或 1。每一个 bit 对应数据表中的一行，并根据该行的取值情况来决定 bit 的取值是 0 还是 1。
 
-Bitmap 索引能够提高指定列的查询效率。如果一个查询条件命中前缀索引列，StarRocks 即可使用[前缀索引](../table_design/Sort_key.md)提高查询效率，快速返回查询结果。但是前缀索引的长度有限，如果想要提高一个非前缀索引列的查询效率，即可以为这一列创建 Bitmap 索引。
+Bitmap 索引能够提高指定列的查询效率。如果一个查询条件命中列，StarRocks 即可使用[前缀索引](./indexes_overview.md#前缀索引)提高查询效率，快速返回查询结果。但是前缀索引的长度有限，如果想要提高一个非前缀索引列的查询效率，即可以为这一列创建 Bitmap 索引。
+
+Bitmap 索引一般适用于高基数列，基于列构建的 Bitmap 索引的选择性高，并且使用 Bitmap 索引后能筛选出的较少的数据行的场景。
```
Comment (amber-create) [inline docs/zh/table_design/indexes/Bitmap_index.md:13]: image
这里不通顺。
是否 -> 能筛选出较少数据行的场景 OR 能筛选出的数据行较少的场景

File: docs/zh/table_design/indexes/Bitmap_index.md [hunk 6]
```diff
@@ -8,11 +8,18 @@ displayed_sidebar: "Chinese"
...
+​
+在 StarRocks中使用 SSB 100G 的测试数据集验证 Bitmap 索引加速查询的效果，测试结果如下：
```
Comment (amber-create) [inline docs/zh/table_design/indexes/Bitmap_index.md:15]: image
-> 100 GB

File: docs/en/table_design/indexes/Bitmap_index.md [hunk 7]
```diff
@@ -2,19 +2,26 @@
...
+
+To evaluate the acceleration performance of Bitmap indexes in StarRocks, queries are performed on a 100 GB SSB dataset in StarRocks. The test results are as follows:
+
+-  when bitmap indexes are added on high-cardinality columns (in this dataset, a noticeable performance improvement can be observed when the cardinality reaches the magnitude of 100,000).
```
Comment (amber-create) [inline docs/en/table_design/indexes/Bitmap_index.md:17]: image
extra space, and "when" should be changed to "When"

File: docs/zh/table_design/indexes/indexes_overview.md [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+
+##### 是否支持修改前缀索引
+
+建表后无法修改排序列（除了主键表外），因此也无法直接修改前缀索引。
+
+如果业务场景中查询特点发生变化，查询条件经常使用前缀字段之外的列，现有的前缀索引无法过滤数据，此时查询性能可能不佳。此时您还可以基于该表创建[同步物化视图](../../using_starrocks/Materialized_view-single_table.md)，并基于其它常用的条件过滤列构建前缀索引，从而提升查询性能。但是注意这样会增加存储空间。
+
+### Ordinal 索引
+
+底层存储数据时，StarRocks 实际上采用列式存储。每一列数据以 Date Page 为单位分块存储，每个 Data Page 大小一般为 64*1024 个字节（data_page_size = 64 * 1024）。每一个列 data page 会对应生成一条 Ordinal 索引项，记录 Data Page 的起始行号等信息。这样 Ordinal 索引提供了通过行号来查找列 Data Page 数据页的物理地址。其他索引查找数据时，最终都要通过 Ordinal 索引查找列 Data Page的位置。
```
Comment (amber-create) [inline docs/zh/table_design/indexes/indexes_overview.md:130]: image
大小写，空格

File: docs/en/table_design/indexes/indexes_overview.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+displayed_sidebar: "English"
+toc_max_heading_level: 5
+---
+
+# Indexes
+
+Indexes can help quickly locate data that matches certain query conditions. To be specific, when an index is built on a table based on some of the table's columns, queries that use these columns as query conditions require scanning only on portions of the table data in order to locate the data that meet the conditions. As such, query efficiency is improved. StarRocks provides a variety of index types, mainly divided into the following two categories:
+
+- StarRocks provides built-in indexes which are automatically created by the system, including Prefix indexes, Ordinal indexes, and ZoneMap indexes.
+- StarRocks also allows users to manually create indexes, including Bitmap indexes and Bloom filter indexes.
```
Comment (amber-create) [inline docs/en/table_design/indexes/indexes_overview.md:11]: image
including: see above comment

File: docs/en/table_design/indexes/indexes_overview.md [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+
+## Built-in indexes
+
+### Prefix indexes
+
+#### Description
+
+Specify one or more columns to comprise the sort key at table creation. The data rows in the table will be sorted based on the sort key and then stored on the disk.
+
+**During data writing, the Prefix index is automatically generated. After the data is sorted according to the specified sort key, every 1024 rows of data are included in one logical data block. An index entry that consists of the values of sort key columns  of the first row data in that logical data block is added to the Prefix index table.**
```
Comment (amber-create) [inline docs/en/table_design/indexes/indexes_overview.md:21]: image
1. extra space
2. data row

File: docs/en/table_design/indexes/indexes_overview.md [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+    phone varchar(16),
+    last_access datetime,
+    credits double
+)
+DUPLICATE KEY(uid, name);
+```
+
+:::note
+
+After creating a Unique Key table, Aggregate table, or Unique Key table, you can use `DESCRIBE <table_name>;` to view its sort key columns. In the returned result, the columns whose `Key` field shows `true` are sort key columns. After creating a Primary Key table, you can use `SHOW CREATE TABLE <table_name>;` to view its sort key columns.  In the returned result, the columns in `ORDER BY` clause are sort key columns.
```
Comment (amber-create) [inline docs/en/table_design/indexes/indexes_overview.md:53]: image
extra space

File: docs/en/table_design/indexes/indexes_overview.md [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+
+:::
+
+Since the maximum length of a Prefix index entry is 36 bytes, the exceeded part will be truncated. Therefore, each entry in the Prefix index of this table is uid (4 bytes) + name (only the first 32 bytes are taken), and the prefix fields are `uid` and `name`.
+
+#### Precaution
+
+- The number of prefix fields cannot exceed 3, and the maximum length of a Prefix index entry is 36 bytes.
+
+- Prefix fields support to be of the xxx type.
```
Comment (amber-create) [inline docs/en/table_design/indexes/indexes_overview.md:63]: image
what "xxx" should be?

File: docs/en/table_design/indexes/indexes_overview.md [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+
+- Among the prefix fields, columns of CHAR/VARCHAR/STRING type can only appear once and must be at the end.
```
Comment (amber-create) [inline docs/en/table_design/indexes/indexes_overview.md:65]: CHAR/VARCHAR/STRING type -> the CHAR, VARCHAR, or STRING type?

File: docs/en/table_design/indexes/indexes_overview.md [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+    | uid         | int         | YES  | true  | NULL    |       |
+    | last_access | datetime    | YES  | true  | NULL    |       |
+    | age         | int         | YES  | false | NULL    |       |
+    | phone       | varchar(16) | YES  | false | NULL    |       |
+    | credits     | double      | YES  | false | NULL    |       |
+    +-------------+-------------+------+-------+---------+-------+
+    6 rows in set (0.00 sec)
+    ```
+
+##### How to design the sort key appropriately to form the Prefix index that can accelerate queries.
```
Comment (amber-create) [inline docs/en/table_design/indexes/indexes_overview.md:84]: remove the period

File: docs/en/table_design/indexes/indexes_overview.md [hunk 15]
```diff
@@ -0,0 +1,150 @@
...
+An analysis of queries and data in business scenarios helps choose appropriate sort key columns and arrange them in a proper order to form a Prefix index, which can significantly improve query performance.
+
+:::tip
+
+Except for Primary Key tables, currently the sort key for other types of tables **cannot be modified** after table creation. Therefore, it is recommended to carefully analyze the characteristics of data and queries to design a suitable sort key.
+
+:::
+
+- The number of sort key columns is generally 3 and is not recommended to exceed 4. A sort key with too many columns can not improve query performance but increase the sorting overhead during data loading.
+- It is recommended to prioritizing columns to form the sort key in the order below:
```
Comment (amber-create) [inline docs/en/table_design/indexes/indexes_overview.md:95]: prioritizing -> prioritize

File: docs/en/table_design/indexes/indexes_overview.md [hunk 16]
```diff
@@ -0,0 +1,150 @@
...
...nce can be significantly improved. And if the filter conditions include the entire prefix of the Prefix index, the query can fully leverage the Prefix index. Of course, as long as the filter conditions include the prefix, though not the entire prefix, the Prefix index can still optimize the query. However, the effect of the Prefix index will be weakened if the length of the prefix included in the filter conditions is too short. Still, take the [Unique Key table](https://chat.openai.com/c/0c47f67a-8103-4ec6-a280-71495f037334#Usage-Guidelines) whose sort key is `(uid,name)` as an example. If the query filter conditions include the entire prefix, such as `select sum(credits) from user_access where uid = 123 and name = 'Jane Smith';`, the query can fully utilize the Prefix index to improve performance.  If the query conditions only include part of the prefix, such as `select sum(credits) from user_access where uid = 123;`, the query can also benefit from the Prefix index to improve performance.  However, if the query conditions do not include the prefix, for example, `select sum(credits) from user_access where name = 'Jane Smith';`, the query can not use the Prefix index to accelerate.
```
Comment (amber-create) [inline docs/en/table_design/indexes/indexes_overview.md:96]: image
extra space

File: docs/en/table_design/indexes/indexes_overview.md [hunk 17]
```diff
@@ -0,0 +1,150 @@
...
+    :::tip
+    
+    However, considering the query characteristics in actual business scenarios, usually, slightly lower cardinality columns are more frequently used as query conditions than high cardinality columns. This is because queries whose filtering is frequently based on high cardinality columns, or even in some extreme scenarios, based on columns with UNIQUE constraints are more like point queries in OLTP databases rather than complex analytical queries in OLAP databases.
+    
+    :::
+  - Also, consider storage compression factors. If the difference in query performance between the order of a low-cardinality column and a high-cardinality one is not readily apparent, placing the low-cardinality column before the high-cardinality one will result in a much higher storage compression rate for the sorted low cardinality column. Therefore, it is recommended to place the low cardinality column in front.
+
+##### Considerations for defining sort key columns at table creation
+
+When defining sort key columns for a Duplicate Key table, an Aggregate table, a Unique Key table, pay attention to the following points:
```
Comment (amber-create) [inline docs/en/table_design/indexes/indexes_overview.md:108]: image
add "or"

File: docs/en/table_design/indexes/indexes_overview.md [hunk 18]
```diff
@@ -0,0 +1,150 @@
...
+
+- The sort key columns must be defined before other columns at table creation.
+- The sort key columns must be the first one or more columns in the table, and the order of the sort key columns must match the sequence of these columns in the table.
+- The data types of the sort key columns do not can be numeric types (excluding DOUBLE and FLOAT), strings, and date types.
```
Comment (amber-create) [inline docs/en/table_design/indexes/indexes_overview.md:112]: image
do not can -> can

File: docs/en/table_design/indexes/indexes_overview.md [hunk 19]
```diff
@@ -0,0 +1,150 @@
...
+
+The Prefix indexes can not be modified directly after table creation (except for Primary Key tables) because the sort key columns can not be changed. If columns other than the prefix fields are frequently used in query filter conditions, the existing Prefix index can not filter data and the query performance may not be ideal. Then you can also create a [synchronous materialized view](../../using_starrocks/Materialized_view-single_table.md) based on this table and choose other columns commonly used as conditional columns to form the Prefix index, which can improve performance for these queries. But note that it will increase storage space.
+
+### Ordinal indexes
+
+StarRocks actually adopts columnar storage in the underlying storage. Data per column is stored in data pages, and the size of each data page's  is generally 64 * 1024 bytes (data_page_size = 64 * 1024). An Ordinal index entry is added at the same time when a data page is generated. The Ordinal index entry contains information such as the starting row number of the data page. In this way, the Ordinal index can locate the physical address of the column Data Page data page by using the row number.
```
Comment (amber-create) [inline docs/en/table_design/indexes/indexes_overview.md:120]: image
extra space

File: docs/en/table_design/indexes/indexes_overview.md [hunk 20]
```diff
@@ -0,0 +1,150 @@
...
+
+### ZoneMap indexes
+
+ZoneMap index stores each data chunk's statistics that include Min (maximum value), Max (minimum value), HasNull (presence of null values), and HasNotNull (presence of non-null values) information. During queries, StarRocks can rapidly evaluate whether these data chunks can be filtered out based on their statistics, reducing the scanned data volume and improving query speed.
+
+<details>
+
+<summary> More information</summary>
+
+Each data chunk can be a segment, or a data page of a column. So correspondingly, two types of  ZoneMap indexes exist: one storing statistics for each Segment and the other for each data page of a column.
```
Comment (amber-create) [inline docs/en/table_design/indexes/indexes_overview.md:132]: image
extra space

File: docs/en/table_design/indexes/indexes_overview.md [hunk 21]
```diff
@@ -0,0 +1,150 @@
...
+
+</details>
+
+## Manually created indexes
+
+If the column in the query condition is not a prefix field, you can manually apply a Bitmap index or Bloom filter index to this column to improve query efficiency.
+
+### [Bitmap indexes](./Bitmap_index.md)
+
+Bitmap indexes are generally suitable for columns with high cardinality. Bitmap indexes are a good choice when  Bitmap indexes can exhibit high selectivity, and its filtering effect (number of data rows filtered by the Bitmap index/total number of data rows) is lower than one in ten thousand.
```
Comment (amber-create) [inline docs/en/table_design/indexes/indexes_overview.md:142]: image
extra space

**Review by amber-create — APPROVED** [review]

**Review by evelynzhaojie — APPROVED** [review]

**Pull request merged**